presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -912,9 +913,36 @@ protected RowExpression visitLikePredicate(LikePredicate node, Context context)
                 return likeFunctionCall(value, call(getSourceLocation(node), "LIKE_PATTERN", functionResolution.likePatternFunction(), LIKE_PATTERN, pattern, escape));
             }
 
+            RowExpression prefixOrSuffixMatch = generateLikePrefix(value, pattern);
+            if (prefixOrSuffixMatch != null) {
+                return prefixOrSuffixMatch;
+            }
+
             return likeFunctionCall(value, call(getSourceLocation(node), CAST.name(), functionAndTypeResolver.lookupCast("CAST", VARCHAR, LIKE_PATTERN), LIKE_PATTERN, pattern));
         }
 
+        private RowExpression generateLikePrefix(RowExpression value, RowExpression pattern)
+        {
+            if ((value.getType() instanceof VarcharType || value.getType() instanceof CharType) && pattern instanceof ConstantExpression) {
+                Object constObject = ((ConstantExpression) pattern).getValue();
+                if (constObject instanceof Slice) {
```

Comment:
Just curious, do we also need to check constant expression pattern's type? Or value is Slice is enough for it?

Reply:
Yeah just to be safe (and also I believe it maybe used for varbinary?l